### ISSUE TITLE
git.latest doesn't checkout submodules

### ISSUE THREAD

**Issue by LoveIsGrief** [saltstack/salt#29009]
Hello,

For the versions salt 2015.8.1 (Beryllium) and salt 2015.5.2 (Lithium) I have run into the problem that `git.latest` doesn't checkout the submodules when asked to do so.

Here's an example configuration to reproduce the problem

```yaml
git_latest:
  git.latest:
    - name: https://github.com/BelledonneCommunications/linphone-android.git
    - target: /tmp/linphone
    - rev: master
    - submodules: True
```

```
vm:
----------
          ID: git_latest
    Function: git.latest
        Name: https://github.com/BelledonneCommunications/linphone-android.git
      Result: True
     Comment: https://github.com/BelledonneCommunications/linphone-android.git cloned to /tmp/linphone
     Started: 16:11:45.449205
    Duration: 43920.872 ms
     Changes:   
              ----------
              new:
                  https://github.com/BelledonneCommunications/linphone-android.git => /tmp/linphone
              revision:
                  ----------
                  new:
                      67d5a2a4de5893dcfdc87f8b5643241766e409af
                  old:
                      None

Summary for vm
------------
Succeeded: 1 (changed=1)
Failed:    0
------------
Total states run:     1
Total run time:  43.921 s
```
```shell
root@vm:/tmp# find /tmp/linphone/submodules/ -name .git
root@vm:/tmp#
```

This ran on a vm:

```
lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 14.04 LTS
Release:        14.04
Codename:       trusty
```

**Comment by jfindlay** [comment]
@LoveIsGrief, thanks for the report.

**Comment by terminalmage** [comment]
Fixed by #29545.
